test/e2e/resize_nodes.go

```diff
@@ -262,16 +262,7 @@ func resizeRC(c clientset.Interface, ns, name string, replicas int32) error {
 func getMaster(c clientset.Interface) string {
 	master := ""
 	switch framework.TestContext.Provider {
-	case "gce":
-		eps, err := c.Core().Endpoints(api.NamespaceDefault).Get("kubernetes")
-		if err != nil {
-			framework.Failf("Fail to get kubernetes endpoinds: %v", err)
-		}
-		if len(eps.Subsets) != 1 || len(eps.Subsets[0].Addresses) != 1 {
-			framework.Failf("There are more than 1 endpoints for kubernetes service: %+v", eps)
-		}
-		master = eps.Subsets[0].Addresses[0].IP
-	case "gke":
+	case "gce", "gke":
 		master = strings.TrimPrefix(framework.TestContext.Host, "https://")
```

Comment:
Hmm this will just break again if we pass a hostname? if we need the master we should probably let the caller decide by passing back a struct with internal/external/hostname, and if we can't find eg: internal ip, leave it blank. to populate the external ip then you'd do this, and try net.Parse to read it as an ip. If it doesn't work, you can assume it's a hostname and put that in leaving the external ip field blank.
Make sense?